docs/sphinx/user_guide/feature/resource.rst

```diff
@@ -185,101 +242,112 @@ Finally, the resource type can be deduced from an execution policy::
+before the third is launched. It also shows copying memory from the device 
+to host on a resource that we described earlier.
     
-First, define two concrete CUDA resources and one host resource:
+First, we define two concrete CUDA resources and one concrete host resource,
+and define an asynchronous CUDA execution policy type:
 
 .. literalinclude:: ../../../../examples/resource-forall.cpp
    :start-after: _raja_res_defres_start
    :end-before: _raja_res_defres_end
    :language: C++
 
-Next, allocate data for two device arrays and one host array:
+Next, we allocate data for two GPU arrays and one host array, all of length 'N':
 
 .. literalinclude:: ../../../../examples/resource-forall.cpp
    :start-after: _raja_res_alloc_start
    :end-before: _raja_res_alloc_end
    :language: C++
 
-Then, Execute a kernel on CUDA stream 1 ``res_gpu1``:
+Then, we launch a GPU kernel on the CUDA stream associated with the resource
+``res_gpu1``, without keeping a handle to the associated event:
 
 .. literalinclude:: ../../../../examples/resource-forall.cpp
    :start-after: _raja_res_k1_start
    :end-before: _raja_res_k1_end
    :language: C++
-    
-and execute another kernel on  CUDA stream 2 ``res_gpu2`` storing a handle to
-an ``Event`` object to a local variable:
+
+Next, we execute another GPU kernel on the CUDA stream associated with the
+resource ``res_gpu2`` and keep a handle to the corresponding event object 
+by assigning it to a local variable ``e``:
 
 .. literalinclude:: ../../../../examples/resource-forall.cpp
    :start-after: _raja_res_k2_start
    :end-before: _raja_res_k2_end
    :language: C++
-    
-The next kernel on ``res_gpu1`` requires that the last kernel on ``res_gpu2`` 
-finish first. Therefore, we enqueue a wait on ``res_gpu1`` that enforces 
-this:
+
+We require that the next kernel we launch to wait for the kernel launched on 
+the stream associated with the resource ``res_gpu2`` to complete. Therefore, 
+we enqueue a wait on that event on the ``res_gpu1`` resource:
 
 .. literalinclude:: ../../../../examples/resource-forall.cpp
    :start-after: _raja_res_wait_start
    :end-before: _raja_res_wait_end
    :language: C++
-    
-Execute the second kernel on ``res_gpu1`` now that the two previous kernels
-have finished:
+
+Now that the second GPU kernel is complete, we launch a second kernel on the
+stream associated with the resource ``res_gpu1``:
 
 .. literalinclude:: ../../../../examples/resource-forall.cpp
    :start-after: _raja_res_k3_start
    :end-before: _raja_res_k3_end
    :language: C++
-    
-We can enqueue a memcpy operation on ``res_gpu1`` to move data from the device 
-to the host:
+
+Next, we enqueue a memcpy operation on the resource ``res_gpu1`` to copy 
+the GPU array ``d_array`` to the host array ``h_array``. **I think we need to
+add a wait call or a device sync call before the memcpy operation since the
+kernel execution is asynchronous --RDH**:
```

Comment:
We don't need a wait call as the memcpy is enqueued on the same stream as the final raja kernel. If the memcpy was enqueued on `res_gpu2` we would need a wait

Reply:
comment removed.